src/coreclr/vm/stackwalk.cpp

```diff
@@ -2720,6 +2711,16 @@ void StackFrameIterator::ProcessIp(PCODE Ip)
     m_crawl.codeInfo.Init(Ip, m_scanFlag);
 
     m_crawl.isFrameless = !!m_crawl.codeInfo.IsValid();
+
+#ifdef TARGET_X86
+    if (m_crawl.isFrameless)
+    {
+        // Optimization: Ensure that we decode GC info header early. We will reuse
+        // it several times.
+        hdrInfo *hdrInfoBody;
+        m_crawl.codeInfo.DecodeGCHdrInfo(&hdrInfoBody);
```

Comment:
I am not sure I understand how this helps. If you would not put it here, wouldn't the next call to `DecodeGCHdrInfo` on the same code info serve the same purpose as this added one?

Reply:
Unfortunately the copy into `m_cachedCodeInfo` breaks the implicit caching, so this was the next best place to trigger it early and ensure that by the time `m_cachedCodeInfo` is populated the cache is also populated.